### PR TITLE
Tabs no longer disappear when scrolling in the extension editor

### PR DESCRIPTION
**Pull request by D8H**
http://gdevelop-storybook.s3-website-us-east-1.amazonaws.com/fix-behavior-scroll/latest/index.html?path=/story/eventsbasedbehavioreditor-eventsbasedbehavioreditordialog--default

It also fixes a typo I made that was forbidding the shared property editor to be displayed.

Maybe the component should be renamed because it only contains one tab now.

### CODE REVIEW

File: newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorEditorDialog.js [hunk 1]
```diff
@@ -14,17 +18,32 @@ type Props = {|
...
+type State = {|
+  currentTab: TabName,
+|};
+
 export default class EventsBasedBehaviorEditorDialog extends React.Component<
   Props,
-  {||}
+  State
 > {
+  state = {
```
Comment (ClementPasteau) [inline newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorEditorDialog.js:29]: can we make this file a functional component while we're at it?
It looks super simple

File: newIDE/app/src/EventsBasedBehaviorEditor/index.js [hunk 2]
```diff
@@ -8,38 +8,21 @@ import * as React from 'react';
...
-  onRenameProperty: (oldName: string, newName: string) => void,
-  onRenameSharedProperty: (oldName: string, newName: string) => void,
-  onTabChanged: () => void,
-|};
-
-type State = {|
-  currentTab: TabName,
 |};
 
 export default class EventsBasedBehaviorEditor extends React.Component<
```
Comment (ClementPasteau) [inline newIDE/app/src/EventsBasedBehaviorEditor/index.js:22]: same for this one

File: newIDE/app/src/EventsBasedObjectEditor/EventsBasedObjectEditorDialog.js [hunk 3]
```diff
@@ -16,10 +21,24 @@ type Props = {|
   onRenameProperty: (oldName: string, newName: string) => void,
 |};
 
+type State = {|
+  currentTab: TabName,
+|};
+
 export default class EventsBasedObjectEditorDialog extends React.Component<
```
Comment (ClementPasteau) [inline newIDE/app/src/EventsBasedObjectEditor/EventsBasedObjectEditorDialog.js:28]: (and this one)

File: newIDE/app/src/EventsBasedObjectEditor/index.js [hunk 4]
```diff
@@ -6,193 +6,115 @@ import { I18n } from '@lingui/react';
...
-  onPropertiesUpdated: () => void,
-  onRenameProperty: (oldName: string, newName: string) => void,
-  onTabChanged: () => void,
-|};
-
-type State = {|
-  currentTab: TabName,
 |};
 
 export default class EventsBasedObjectEditor extends React.Component<
```
Comment (ClementPasteau) [inline newIDE/app/src/EventsBasedObjectEditor/index.js:20]: (and this one)

File: newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorEditorDialog.js [hunk 5]
```diff
@@ -14,55 +18,84 @@ type Props = {|
...
-      </Dialog>
-    );
-  }
+      )}
+      {currentTab === 'behavior-properties' && (
+        <EventsBasedBehaviorPropertiesEditor
+          allowRequiredBehavior
+          project={project}
+          properties={eventsBasedBehavior.getPropertyDescriptors()}
+          onPropertiesUpdated={() => {}}
```
Comment (ClementPasteau) [inline newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorEditorDialog.js:86]: is this property useful anymore? (same for other component below)

**Comment by D8H** [comment]
I did a manual check on the last commit, the scroll works correctly when properties are added or removed.

**Pull request merged**